docs/en/tutorials/customize_dataset.md

```diff
@@ -291,200 +321,43 @@ dataset_A_train = dict(
-For example, assume the `classes.txt` contains the name of classes as the following.
-
-```
-person
-bicycle
-car
-```
-
-Users can set the classes as a file path, the dataset will load it and convert it to a list automatically.
-
-```python
-classes = 'path/to/classes.txt'
-data = dict(
-    train=dict(classes=classes),
-    val=dict(classes=classes),
-    test=dict(classes=classes))
+train_dataloader = dict(
+    dataset=dict(
+        metainfo=dict(CLASSES=classes))
+    )
+val_dataloader = dict(
+    dataset=dict(
+        metainfo=dict(CLASSES=classes))
+    )
+test_dataloader = dict(
+    dataset=dict(
+        metainfo=dict(CLASSES=classes))
+    )
 ```
 
 **Note**:
 
-- Before MMDetection v2.5.0, the dataset will filter out the empty GT images automatically if the classes are set and there is no way to disable that through config. This is an undesirable behavior and introduces confusion because if the classes are not set, the dataset only filter the empty GT images when `filter_empty_gt=True` and `test_mode=False`. After MMDetection v2.5.0, we decouple the image filtering process and the classes modification, i.e., the dataset will only filter empty GT images when `filter_empty_gt=True` and `test_mode=False`, no matter whether the classes are set. Thus, setting the classes only influences the annotations of classes used for training and users could decide whether to filter empty GT images by themselves.
-- Since the middle format only has box labels and does not contain the class names, when using `CustomDataset`, users cannot filter out the empty GT images through configs but only do this offline.
-- Please remember to modify the `num_classes` in the head when specifying `classes` in dataset. We implemented [NumClassCheckHook](https://github.com/open-mmlab/mmdetection/blob/master/mmdet/datasets/utils.py) to check whether the numbers are consistent since v2.9.0(after PR#4508).
-- The features for setting dataset classes and dataset filtering will be refactored to be more user-friendly in the future (depends on the progress).
+- Before MMDetection v2.5.0, the dataset will filter out the empty GT images automatically if the classes are set and there is no way to disable that through config. This is an undesirable behavior and introduces confusion because if the classes are not set, the dataset only filter the empty GT images when `filter_empty_gt=True` and `test_mode=False`. After MMDetection v2.5.0, we decouple the image filtering process and the classes modification, i.e., the dataset will only filter empty GT images when `filter_cfg=dict(filter_empty_gt=True)` and `test_mode=False`, no matter whether the classes are set. Thus, setting the classes only influences the annotations of classes used for training and users could decide whether to filter empty GT images by themselves.
+- When using `BaseDataset` in MMEngine, users cannot filter images without GT by modifying the configuration, but it can be solved in an offline way.
```

Comment:
```suggestion
- When directly using `BaseDataset` in MMEngine, users cannot filter images without GT by modifying the configuration, but it can be solved in an offline way.
```